src/logging.ts

```diff
@@ -0,0 +1,25 @@
+import { Application, Request, Response } from "express";
+import morgan from "morgan";
+import { Dict } from "./types";
+
+export const initialiseLogging = (app: Application) => {
+    const customFormat = (
+        tokens: Dict<(req: Request, res: Response, header?: string) => string>,
```

Comment:
Not able to parse this - what does 'tokens' refer to?

Reply:
It refers to the bits of information relating to a request/response pair which can be logged by `morgan` (the logging package). As you can see from the type, these are actually functions, and this whole `customFormat` method is passed to `morgan` to tell it how to log requests.

Reply:
It's confusing, but I guess all is explained in the morgan docs.